packages/coin98/src/adapter.ts

```diff
@@ -0,0 +1,169 @@
+import {
+  EventEmitter,
+  pollUntilReady,
+  WalletAccountError,
+  WalletAdapter,
+  WalletAdapterEvents,
+  WalletNotConnectedError,
+  WalletNotFoundError,
+  WalletPublicKeyError,
+  WalletSignatureError,
+  WalletNotInstalledError
+} from '@solana/wallet-adapter-base';
+import { PublicKey, Transaction } from '@solana/web3.js';
+import bs58 from 'bs58'
+
+
+interface SignTransactionResponse {
+  signature: string;
+  publicKey: string;
+}
+
+interface RequestParams {
+  method: string;
+  params: string | string[] | unknown
+}
+
+interface Coin98SolProvider{
+  isCoin98?: boolean;
+  signTransaction: (transaction: Transaction) => Promise<Transaction>;
+  isConnected: () => boolean,
+  connect: ()=> Promise<string[]>,
+  disconnect: ()=> Promise<void>,
+  request: (params: RequestParams) => Promise<SignTransactionResponse>
+}
+
+interface Coin98Wallet{
+  sol?: Coin98SolProvider,
+  provider: unknown
+}
+
+interface Coin98Window extends Window {
+  coin98?: Coin98Wallet;
+}
+
+declare const window: Coin98Window;
+
+export interface Coin98WalletAdapterConfig {
+  pollInterval?: number;
+  pollCount?: number;
+}
+
+export class Coin98WalletAdapter extends EventEmitter<WalletAdapterEvents> implements WalletAdapter {
+  private _connecting: boolean;
+  private _wallet: Coin98SolProvider | null;
+  private _publicKey: PublicKey | null;
+
+  constructor(config: Coin98WalletAdapterConfig = {}) {
+      super();
+      this._connecting = false;
+      this._wallet = null;
+      this._publicKey = null;
+      this.signTransaction = this.signTransaction.bind(this);
+      this.signAllTransactions = this.signAllTransactions.bind(this);
```

Comment:
It shouldn't be necessary to bind these methods -- `signAllTransactions` uses an arrow function to invoke `signTransaction`

Reply:
removed.